### ISSUE TITLE
[cssom] definition of getComputedStyle refers to resolved values at the wrong time

### ISSUE THREAD

**Issue by dbaron** [w3c/csswg-drafts#6144]
The [definition of `getComputedStyle()`](https://drafts.csswg.org/cssom/#dom-window-getcomputedstyle) currently has the following as part of its step 5:

> set decls to a list of all longhand properties that are supported CSS properties, in lexicographical order, with the value being the resolved value computed for obj using the style rules associated with doc.

and then the following as part of its step 6:

> Return a live CSS declaration block with the following properties:
>> **computed flag**
>>> Set.

The way this references the resolved values (despite the "live CSS declaration block") seems to imply that the access to the resolved values happens at the time that `getComputedStyle` is called, when in fact the resolved values are computed at the time that `getPropertyValue` or equivalent accessors are called on the live declaration.

It seems to me that this would be clearer if there were something in the definition of [`getPropertyValue`](https://drafts.csswg.org/cssom/#dom-cssstyledeclaration-getpropertyvalue) that considered the **computed flag**.

**Comment by emilio** [comment]
> The way this references the resolved values (despite the "live CSS declaration block") seems to imply that the access to the resolved values happens at the time that getComputedStyle is called, when in fact the resolved values are computed at the time that getPropertyValue or equivalent accessors are called on the live declaration.
>
> It seems to me that this would be clearer if there were something in the definition of getPropertyValue that considered the computed flag.

Yeah, I agree with this. This would also be closer to how implementations work.

**Comment by Loirooriol** [comment]
BTW, not just the values, the number of declarations is also live (e.g. if you remove the element, `.length` will become 0).
